Assets/Scripts/Terrain/Erosion/ErosionUtils.cs

```diff
@@ -0,0 +1,205 @@
+        /// <param name="waterFactor">Amount of water in droplet</param>
+        /// /// <param name="parameters">Erosion parameters for controlling how erosion works</param>
+        /// <returns>The computed capacity of the droplet or Minimum capacity fi it is less than
+        /// than the computed value.</returns>
+        public static float ComputeCapacity(float deltaH, float velocity, float waterFactor, HydroErosionParams parameters) {
+            float slopeFactor = Mathf.Max(Mathf.Abs(deltaH), parameters.minSlope);
+            float velFactor = Mathf.Max(1, parameters.includeVelocity ? velocity : 1);
+            float capacity = slopeFactor * velFactor * waterFactor * parameters.sedimentCapacityFactor;
+            return Mathf.Max(capacity, parameters.minCapacity);
+        }
+
+        /// <summary>
+        /// Evaluate how much sediment is deposited and deposit that much
+        /// sediment based on the droplets current state.
+        /// </summary>
+        /// <param name="deltaH">Change in height since last move</param>
+        /// <param name="sediment">Amount of sediment current held in the droplet</param>
+        /// <param name="capacity">Capacity of the current droplet</param>
+        /// <param name="pos">Position of the current droplet</param>
+        /// <param name="map">Map to read and make changes to</param>
+        /// <param name="parameters">Erosion parameters for controlling how erosion works</param>
+        /// <returns>Actual amount of sediment deposited</returns>
+        public static float DepositSediment(float deltaH, float sediment, float capacity,
+            Vector2 pos, HeightMap map, HydroErosionParams parameters) {
+            // Deposit all sediment if moving uphill (but not more than the size of the pit)
+            float slopeBasedDeposit = Mathf.Min (deltaH, sediment);
+            // Deposit proportion of sediment based on capacity
+            float capacityBasedDeposit = (sediment - capacity) * parameters.depositionRate;
+            // Select slope or capacity based on if moving uphill
+            float amountToDeposit = (deltaH > 0) ? slopeBasedDeposit : capacityBasedDeposit;
+            return ErosionUtils.Deposit(map, pos, amountToDeposit);
+        }
+
+        /// <summary>
+        /// Erodes a height map at a given position using a erosion brush.
+        /// </summary>
+        /// <param name="map">Map to change</param>
+        /// <param name="pos">Position of droplet on map</param>
+        /// <param name="amountToErode">Total amount to be removed</param>
+        /// <param name="radius">Radius of the brush</param>
+        /// <param name="brush">Brush to use when applying erosion</param>
+        /// <returns>The total amount of soil eroded (might be slightly less than amountToErode</returns>
+        public static float Erode(HeightMap map, Vector2 pos, float amountToErode, int radius, float[,] brush) {
```

Comment:
You could make these functions static extensions of the HeightMap class. This would make using them possibly more seamless in practice. ```map.Erode(*paramaters*)``` > ```ErosionUtils.Erode(maps, *parmaters*)```

Reply:
Wow, Didn't realize that was a thing. I added the feature and it looks much better.